### ISSUE TITLE
Found many accessibility violations with quick accessibility check

### ISSUE THREAD

**Issue by akash07k** [SSilence/selfoss#1132]
First, let's know that what's the accessibility?
https://www.w3.org/WAI/fundamentals/accessibility-intro/
https://webaim.org/intro/
So, accessibility is really the important part of everything in todays world and it is really needed.
I've evaluated the basic accessibility of the selfoss driven website and there are many accessibility issues found with the quick accessibility pass itself.
I've done this tast pass with the opensource tool by microsoft which is named: 'accessibility insights for web'
https://accessibilityinsights.io/docs/en/web/overview
https://chrome.google.com/webstore/detail/accessibility-insights-fo/pbjjkligggfmakdaogkfomddhfmpjeni
Please find the attached test pass results with more details

[AutomatedChecks_20190916_Readerotica1281.docx](https://github.com/SSilence/selfoss/files/3616094/AutomatedChecks_20190916_Readerotica1281.docx)

**Comment by jtojnar** [comment]
Thanks for reporting. Accessibility is something I am always trying to improve in selfoss, most of the UI should be navigable by keyboard and I also tried using it blindfolded (though I only have access to Orca screen reader). Overall I would say it is usable, though there are still many small annoyances in need of a fix.

I fixed some of the issues in #1133 but will leave the contrast issues to a design brush up later in the cycle.

**Comment by akash07k** [comment]
@jtojnar 
Thanks a lot for your efforts bro.
Actually, there are couple of more things, such as, the landmarks should be correctly implemented, skip links should be properly implemented and all the shortcut keys should be exposed by using accesskey or aria attributes to the screen reader users.
Also, I've read in #1028 
That you need some help with aria live regions? so, please let me know that how can I help? I'll try to become familiarise with the selfoss code myself and will try to fix somethings alongside. BTW, selfoss is the best!

**Comment by akash07k** [comment]
Today I'll pull all your patches and will provide more feedback on this.

**Comment by akash07k** [comment]
And yes, we should try to expose the control type of every control by using role="valid role" if it is defined in a div.
If you have the time, then you can go through these links:

https://support.siteimprove.com/hc/en-gb/articles/115003457351-How-skip-links-can-help-with-website-accessibility
https://webaim.org/techniques/skipnav/
https://accessibility.oit.ncsu.edu/it-accessibility-at-nc-state/developers/accessibility-handbook/aria-landmarks/
https://accessibility.oit.ncsu.edu/using-aria-landmarks-a-demonstration/
https://dequeuniversity.com/assets/html/jquery-summit/html5/slides/landmarks.html

**Comment by akash07k** [comment]
We can add aria-live regions for the instances where we refresh the sources, add the sources, edit the sources etc.
Also, the aria-expanded state should be associated with the every feed heading because every feed heading is collapsable and expandable.
With this, screen reader user will be able to identify that the feed is expanded or collapsed.
And, since, the feed heading is always clickable, the role="link" should be used on that along with aria-expanded"

**Comment by jtojnar** [comment]
Are any of these still relevant?

**Comment by akash07k** [comment]
I will run the accessibility check today again and will let you know
